include/SugarObjects/templates/basic/Basic.php

```diff
@@ -134,24 +179,23 @@ public function getEmailAddressConfirmOptIn($emailField)
      * @throws RuntimeException
      * @throws InvalidArgumentException
      */
-    private function getEmailAddressId($emailField)
+    protected function getEmailAddressId($emailField)
     {
-        $this->getEmailAddressValidateArguments($emailField);
```

Comment:
Why should remove the validation?

Reply:
I removed this method because I was having trouble with the emails module. getEmailAddressValidateArguments assumes that the filed will end in a number like email1, email2, ... etc. Which works great for Accounts, Contacts and Leads.

However, the Emails module stores the email address in email_address field. Which will not pass this validation.

So ... we need to come up with something which better handles these constraints. The real issue is where this code is situated. The responsibility for the email1, should have been implemented as a separate class, via an interface, or perhaps it we should employ a trait. Ideally this would be in the EmailAddresses module instead of the SugarBean. Since we cannot time travel back and change the original problem. We will have to accept the existing code in SugarBean.

We could use the inheritance tree. What If we changed the method to be protected instead. Then override the method. Should these methods existing in the basic type? Should this in company and person instead?

Reply:
if the `email_address` field also valid, why don't you just add it to the validation instead removing the validation at all?